nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/kinesis/stream/pause/TestStandardRecordProcessorBlocker.java

```diff
@@ -0,0 +1,114 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.aws.kinesis.stream.pause;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.function.Supplier;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TestStandardRecordProcessorBlocker {
+
+    @Test
+    public void testBlockAndUnblock() {
+        final TestThreadInspector blockerInspector = new TestThreadInspector();
+
+        final StandardRecordProcessorBlocker recordProcessorBlocker = StandardRecordProcessorBlocker.create();
+
+        recordProcessorBlocker.block();
+        final Thread thread = new Thread(createRunnableWithInspector(recordProcessorBlocker, blockerInspector));
+        thread.start();
+
+        blockerInspector.awaitAwaited();
+        assertTrue(thread.isAlive());
+
+        recordProcessorBlocker.unblock();
+        blockerInspector.awaitFinished();
+    }
+
+    @Test
+    public void testNoBlock() {
+        final TestThreadInspector blockerInspector = new TestThreadInspector();
+
+        final StandardRecordProcessorBlocker recordProcessorBlocker = StandardRecordProcessorBlocker.create();
+
+        recordProcessorBlocker.unblock();
+        final Thread thread = new Thread(createRunnableWithInspector(recordProcessorBlocker, blockerInspector));
+        thread.start();
+
+        blockerInspector.awaitFinished();
+    }
+
+    @Test
+    public void testBlock() {
+        final TestThreadInspector blockerInspector = new TestThreadInspector();
+
+        final StandardRecordProcessorBlocker recordProcessorBlocker = StandardRecordProcessorBlocker.create();
+        recordProcessorBlocker.block();
+        final Thread thread = new Thread(createRunnableWithInspector(recordProcessorBlocker, blockerInspector));
+        thread.start();
+
+        blockerInspector.awaitAwaited();
+        assertTrue(thread.isAlive());
+
+        recordProcessorBlocker.unblock();
+    }
+
+    private static Runnable createRunnableWithInspector(final StandardRecordProcessorBlocker recordProcessorBlocker, TestThreadInspector blockerInspector) {
+        return () -> {
+            try {
+                blockerInspector.onPauseAwaited();
+                recordProcessorBlocker.await();
+            } catch (final InterruptedException e) {
+                throw new RuntimeException(e);
+            } finally {
+                blockerInspector.onPauseFinished();
+            }
+        };
+    }
+
+    private static class TestThreadInspector {
+        private boolean onPauseAwaited = false;
+        private boolean onPauseFinished = false;
+
+        public void onPauseAwaited() {
+            onPauseAwaited = true;
+        }
+
+        public void onPauseFinished() {
+            onPauseFinished = true;
+        }
```

Comment:
It looks like these variables and methods should be updated to reflect the `Blocker` naming.